### DISCUSSION TITLE
[Kamal 2] Redirect http to https

### DISCUSSION THREAD

**Discussion by JoshuaBellew** [basecamp/kamal#1118] [Q&A] [answered]
Hi all, 

I've just started using Kamal 2 to deploy to hetzner and it went pretty smoothly!

Is there any way I can set up a http to https redirect via the config file? Port 80 -> Port 443.

Many thanks

**Comment by dhh** [comment, marked as answer]
Kamal doesn't have anything for that. You'll need to do it in your app or if you put Cloudflare or whatever in front.

**Reply by JoshuaBellew** [reply]
Thanks David! Appreciate your open source work with Kamal!

**Reply by rahearn** [reply]
Is this still true? I'm trying to serve one route in my app via both http and https and since moving from heroku to a kamal setup I'm getting 301 redirects to https on every route and the request never appears to get to my app. I have `config.ssl_options = {redirect: {exclude: ->(request) { %r{^/path/regex$} =~ request.path }}}` still setup that's working fine when hosting in heroku.

**Reply by rahearn** [reply]
For anyone finding this later, it does look like kamal-proxy redirects all http traffic to https when it is setup to serve tls:

https://github.com/basecamp/kamal-proxy/blob/dcde7ca7768e851a6f13c038963a9226c93046b0/internal/server/service.go#L354-L357
